testsuite/tests/lib-format/with_size.ml

```diff
@@ -0,0 +1,60 @@
+(*TEST
+expect;
+*)
+
+let () = Format.set_geometry ~margin:10 ~max_indent:9
+let test fmt =
+  Format.printf "@[";
+  Format.kfprintf (fun ppf -> Format.fprintf ppf "@]@.-----------@.") Format.std_formatter
+    fmt;;
+[%%expect {|
+val test : ('a, Format.formatter, unit, unit) format4 -> 'a = <fun>
+|}]
+
+let () = test "foo@ @<1000>bar"
+let () = test "foo@ @<1000>%s" "bar"
+let () = test "foo@ @<1000>%a" Format.pp_print_string "bar"
```

Comment:
It might be good to have another test case similar to this one, but such as
```
t () = test "foo@ @<1000>%a" (fun ppf -> Format.fprintf ppf "@[bar@]")
```
to illustrate that using boxes will defeat `with_size`.